### PR TITLE
feat(nullable/option): add nullable.iter and option.iter

### PR DESCRIPTION
**Pull request by rmjohnson**

### CODE REVIEW

File: src/Option/Option.ts [hunk 1]
```diff
@@ -142,6 +142,29 @@ export const map = <A extends NonNullish, B extends NonNullish>(
...
+ *
+ * pipe(
+ *     Option.none,
+ *     Option.iter((s: string) => fooCallback(s))
+ * ) // fooCallback is not called
+ */
+export const iter = <A extends NonNullish>(f: (a: A) => void) =>
+    match<A, void>({
+        some: a => f(a),
+        none: () => {},
```
Comment (rmjohnson) [inline src/Option/Option.ts]: Let me know if there's a better way to do this such that we aren't creating a useless arrow function.

File: src/Option/Option.ts [hunk 2]
```diff
@@ -142,6 +142,29 @@ export const map = <A extends NonNullish, B extends NonNullish>(
...
+    })
```
Comment (taylorn16) [inline src/Option/Option.ts]: does this work?

```suggestion
    match<A, void>({
        some: f,
        none: void 0,
    })
```

File: src/Option/Option.ts (hunk 1 above)
Comment (taylorn16) [reply src/Option/Option.ts]: See https://github.com/fp-toolkit/fp-toolkit/pull/72/files#r1735068930. If `void 0` doesn't work, try `undefined` as well

File: src/Option/Option.ts (hunk 2 above)
Comment (rmjohnson) [reply src/Option/Option.ts]: Beautiful 😍 I tried `void` but was missing the `0`.

File: src/Option/Option.ts (hunk 1 above)
Comment (rmjohnson) [reply src/Option/Option.ts]: `void 0` worked great :tada:

File: src/Option/Option.ts (hunk 2 above)
Comment (rmjohnson) [reply src/Option/Option.ts]: Updated!

File: src/Option/Option.ts [hunk 3]
```diff
@@ -142,6 +142,29 @@ export const map = <A extends NonNullish, B extends NonNullish>(
         none: none,
     })
 
+/**
+ * Execute an arbitrary side-effect function if `Some`.
```
Comment (taylorn16) [inline src/Option/Option.ts:146]: 🎨 Might be worth noting in the remarks that this is a terminal function in a `pipe`line, i.e. this does not pass any value through (and probably same for `Nullable.iter`)

Comment (rmjohnson) [reply src/Option/Option.ts:146]: Updated!

**Review by taylorn16 — APPROVED** [review]
Thank you!

**Pull request merged**